static/js/Header.jsx

```diff
@@ -393,6 +401,346 @@ SearchBar.propTypes = {
   hideHebrewKeyboard: PropTypes.bool,
 };
 
+const SearchSuggestion = ({ item }) => {
+
+  const _type_icon_map = {
+      "Collection": "collection.svg",
+      "AuthorTopic": "iconmonstr-pen-17.svg",
+      "TocCategory": "iconmonstr-view-6.svg",
+      "PersonTopic": "iconmonstr-hashtag-1.svg",
+      "Topic": "iconmonstr-hashtag-1.svg",
+      "ref": "iconmonstr-book-15.svg",
+      "search": "iconmonstr-magnifier-2.svg",
+      "Term": "iconmonstr-script-2.svg",
+      "User": "iconmonstr-user-2%20%281%29.svg"
+    }
+
+  const _type_icon = function(item) {
+    if (item.type === "User" && item.pic !== "") {
+      return item.pic;
+    } else {
+      return `/static/icons/${_type_icon_map[item.type]}`;
+    }
+  }
+
+  const isHebrew = Sefaria.hebrew.isHebrew(item.label);
+
+  return (
+     <div className="ui-menu-item"
+        data-item-autocomplete={item}
+        className={`
+          hebrew-result ${!!isHebrew ? 'hebrew-result' : ''} 
+          english-result ${!isHebrew ? 'english-result' : ''}
+        `}>
+      <img alt={item.type}
+           className={`ac-img-${item.type === "User" && item.pic === "" ? "UserPlaceholder" : item.type}`}
+           src={_type_icon(item)} />
+       <a href={item.url} role="option" data-type-key={`${item.type}-${item.key}`}>
+        {item.label}
+      </a>
+    </div>  );
+};
+
+const InputBox = ({getInputProps, suggestions, highlightedIndex, ...props}) => {
+    const [searchFocused, setSearchFocused] = useState(false);
+
+    useEffect(() => {
+      showVirtualKeyboardIcon(false); // Initially hide the virtual keyboard icon
+    }, []);
+   const { onBlur, onKeyDown, ...otherDownShiftProps } = getInputProps();
+
+   const clearSearchBox = function () {
+     getInputProps().onChange({ target: { value: '' } });
+  }
+   const submitSearch = (query) => {
```

Comment:
This function is long and I think it would be better if it was pulled out of the InputBox component. The function can return an object of the form:

```
{ type: str, id: str }
```

Where `type` is either `Ref`, `Topic`, `Person`, `Collection`, `TocCategory` or `Search`
`id` is the unique ID of the search result. For `Ref`, that's the ref itself. For `Topic` that's the `topic_slug`, for `Search` that's the query itself. For all other types it's `key`

I believe this way we can separate the state management from the search logic.